pmd-core/src/main/java/net/sourceforge/pmd/properties/PropertyFactory.java

```diff
@@ -172,6 +172,38 @@ public static RegexPropertyBuilder regexProperty(String name) {
     }
 
 
+    /**
+     * Returns a builder for a boolean property. The boolean is parsed from
+     * the XML using {@link Boolean#valueOf(String)}, i.e. the only truthy
+     * value is the string "true", and all other string values are falsy.
+     *
+     * @param name Name of the property to build
+     *
+     * @return A new builder
+     *
+     * @see #booleanProperty(String, boolean)
+     */
+    public static GenericPropertyBuilder<Boolean> booleanProperty(String name) {
+        return new GenericPropertyBuilder<>(name, ValueParserConstants.BOOLEAN_PARSER, Boolean.class);
+    }
+
+
+    /**
+     * Returns a builder for a boolean property, with a default value
+     * already specified. This is shorthand for {@code booleanProperty(name).defaultValue(defaultValue)}
+     *
+     * @param name         Name of the property to build
+     * @param defaultValue Default value of the property
+     *
+     * @return A new builder
+     *
+     * @see #booleanProperty(String)
+     */
+    public static GenericPropertyBuilder<Boolean> booleanProperty(String name, boolean defaultValue) {
```

Comment:
Should we really provide this shorthand? We don't have it for the other properties - so having this for boolean only looks inconsistent.
I'd just remove this shorthand, since you still need to call ".desc(...)" and ".build()" on the returned builder, it only saves you "defaultValue".

Reply:
You're probably right - I didn't even convert those old properties consistently